site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,152 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import FormHelperText from "@mui/material/FormHelperText";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField, { type TextFieldProps } from "@mui/material/TextField";
+import { useState, type FC, useEffect } from "react";
+import {
+  type TimeUnit,
+  durationInDays,
+  durationInHours,
+  suggestedTimeUnit,
+} from "utils/time";
+
+type DurationFieldProps = Omit<TextFieldProps, "value" | "onChange"> & {
+  // Value is in ms
+  value: number;
+  onChange: (value: number) => void;
+};
+
+type State = {
+  unit: TimeUnit;
+  // Handling empty values as strings in the input simplifies the process,
+  // especially when a user clears the input field.
+  durationFieldValue: string;
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const { value: parentValue, onChange, helperText, ...textFieldProps } = props;
+  const [state, setState] = useState<State>(() => initState(parentValue));
```

Comment:
The state is still small enough that I think either approach works well, but just to throw out an alternative approach, I'm wondering if `useReducer` might be a handy swap-in here, especially since the two properties always need to update together